doc/musig-spec.mediawiki

```diff
@@ -60,7 +62,120 @@ The following conventions are used, with constants as defined for [https://www.s
 ** The function ''point(x)'', where ''x'' is a 32-byte array ("x-only" serialization), returns ''lift_x(int(x))''. Fail if ''lift_x'' fails.
 ** The function ''pointc(x)'', where ''x'' is a 33-byte array (compressed serialization), sets ''P = lift_x(int(x[1:33]))'' and fails if that fails. If ''x[0] = 2'' it returns ''P'' and if ''x[0] = 3'' it returns ''-P''. Otherwise, it fails.
 ** The function ''hash<sub>tag</sub>(x)'' where ''tag'' is a UTF-8 encoded tag name and ''x'' is a byte array returns the 32-byte hash ''SHA256(SHA256(tag) || SHA256(tag) || x)''.
+* Other:
+** Tuples are written by listing the elements within parentheses and separated by commas. For example, ''(2, 3, 1)'' is a tuple.
+
+=== Tweaking ===
+
+Input:
+* ''P'': a point
+* The tweak ''t'': an integer with ''0 &le; t < n ''
+
+The algorithm ''OrdinaryTweak(P, t)'' is defined as:
+* Return P + t⋅G
+
+The algorithm ''XonlyTweak(P, t)'' is defined as:
+* Return set_even_y(P) + t⋅G
+
+=== Negation Of The Secret Key When Signing ===
+
+In order to produce a partial signature for an x-only public key that is an aggregate of x-only keys and x-only or ordinarily tweaked, the ''[[#Sign negation|Sign]]'' algorithm may need to negate the secret key during the signing process.
+
+<poem>
+The following public keys arise as intermediate steps in the MuSig protocol:
+• ''P<sub>i</sub>'' as computed in ''KeyAggInternal'' is the point corresponding to the ''i''-th signer's x-only public key. Defining ''d'<sub>i</sub>'' to be the ''d' '' value as computed in the ''Sign'' algorithm of the ''i''-th signer, we have
+    ''P<sub>i</sub> = set_even_y(d'<sub>i</sub>⋅G) ''.
+• ''Q<sub>0</sub>'' as defined in ''Tweak'' is the aggregate public key computed in ''KeyAggInternal'' as
```

Comment:
defined in which Tweak? "OrdinaryTweak or XonlyTweak"?
But even then I can't fully parse this. There's no Q_0 defined in either of the tweak algorithms..

edit: Ah you mean the algorithm below. So this was very confusing to me. Maybe point out that it can be found below. But I think then it's still not optimal for the reader because this here requires that one has read later parts. Maybe it's better to move this entire "correctness proof"/ explanation below the spec algos?

Reply:
I moved this section below the specification section. That seems moved reasonable because it's content not immediately relevant to someone implementing the specification.

Reply:
I think it's much better now. :)